src/services/services.ts

```diff
@@ -1974,20 +1972,31 @@ namespace ts {
                         node.parent.kind === SyntaxKind.ExternalModuleReference ||
                         isArgumentOfElementAccessExpression(node) ||
                         isLiteralComputedPropertyDeclarationName(node)) {
-
-                        const text = (<LiteralExpression>node).text;
-                        nameTable.set(text, nameTable.get(text) === undefined ? node.pos : -1);
+                        setNameTable((<LiteralExpression>node).text, node);
                     }
                     break;
-                default:
+                case SyntaxKind.TypeOperator:
+                    setNameTable(tokenToString((node as ts.TypeOperatorNode).operator), node);
                     forEachChild(node, walk);
-                    if (node.jsDoc) {
-                        for (const jsDoc of node.jsDoc) {
-                            forEachChild(jsDoc, walk);
+                    break;
+                default:
+                    if (isTypeKeyword(node.kind)) {

```

Comment:
does this buy you much really.. like `string`, `number` will likely be in each file, might as well do the search.